include/parameter.h

```diff
@@ -0,0 +1,307 @@
+            inheritor->str() + ") with inheritors of its own; inheritance may not be nested");
+        if(_inheritors == nullptr) _inheritors = std::make_unique<SetC>();
+        _inheritors->insert(inheritor);
+    }
+    inline void add_modifier(std::shared_ptr<C> modifier) { 
+        if(modifier == nullptr) throw std::runtime_error("Can't add null modifier to " + this->str());
+        if(_modifiers == nullptr) _modifiers = std::make_unique<SetC>();
+        _modifiers->insert(modifier);
+    }
+
+    static const std::string get_desc() { return _desc_t<C>(0); }
+    static constexpr T get_default() { return _default_t<C>(0); }
+    bool get_fixed() const { return !_free; }
+    bool get_free() const { return _free; }
+    std::shared_ptr<const C> get_inheritee() const { return _inheritee_ptr; }
+    inline const SetC get_inheritors() const { return _inheritors == nullptr ? SetC() : *_inheritors; }
+    std::string get_label() const { return _label; }
+    inline const Limits<T> & get_limits() const { return _limiter->limits; }
+    static constexpr T get_min() { return _min_t<C>(0); }
+    static constexpr T get_max() { return _max_t<C>(0); }
+    inline const SetC get_modifiers() const { return _modifiers == nullptr ? SetC() : *_modifiers; }
+    static inline const std::string get_name() { return _name_t<C>(0); }
+    inline const Transform<T> & get_transform() const { return _transformer->transform; }
+    inline T get_transform_derivative() const { return this->get_transform().derivative(this->get_value()); }
+    static inline const std::string get_type_name() { return std::string(type_name<C>()); }
+    inline const std::shared_ptr<const Unit> get_unit() const { return _unit_ptr; }
+    T get_value() const { return _value; }
+    T get_value_transformed() const { return _value_transformed; }
+
+    std::shared_ptr<C> ptr() { return this->shared_from_this(); }
+
+    inline void remove_inheritor(std::shared_ptr<C> inheritor) {
+        if(inheritor == nullptr) throw std::runtime_error("Can't remove null inheritor from " + this->str());
+        if(_inheritors == nullptr) throw std::runtime_error("Can't remove inheritor from " + this->str() + " with no inheritors");
+        _inheritors->erase(inheritor); 
+    }
+
+    inline void remove_modifier(std::shared_ptr<C> modifier) {
+        if(modifier == nullptr) throw std::runtime_error("Can't remove null modifier from " + this->str());
+        if(_modifiers == nullptr) throw std::runtime_error("Can't remove modifier from " + this->str()
+            + " with no modifiers");
+        _inheritors->erase(modifier); 
+    }
+
+    void set_fixed(bool fixed) { _free = !fixed; }
+    void set_free(bool free) {
+        if(_inheritee_ptr != nullptr) "Can't set this=" + this->str() + " free while it inherits from "
+            "inheritee = " + _inheritee_ptr->str();
+        _free = free;
+    }
+    void set_label(std::string label) { _label = label; }
```

Comment:
```suggestion
    void set_label(std::string label) { _label = std::move(label); }
```
Or pass by const reference.